source/MaterialXContrib/MaterialXNode/MaterialXData.h

```diff
@@ -1,34 +1,73 @@
 #ifndef MATERIALX_DATA_H
 #define MATERIALX_DATA_H
 
+/// @file
+/// MaterialX Data wrapper
+
 #include <MaterialXCore/Document.h>
 #include <MaterialXGenShader/GenContext.h>
-
-#include "../OGSXMLFragmentWrapper.h"
+#include <MaterialXContrib/OGSXMLFragmentWrapper.h>
 
 #include <maya/MString.h>
 
-#include <vector>
-
+/// @class MaterialXData
+/// Wrapper for MaterialX associated data. 
+///
+/// Keeps track of an element to render and it's associated document.
+///
+/// Can optionally create and cache an OGS XML wrapper instance 
+/// which wraps up the interface and shader code shader code generated based 
+/// on the specified element to render.
+/// Currently only code for GLSL is generated.
+///
 struct MaterialXData
 {
-	MaterialXData(const std::string& materialXDocument, const std::string& elementPath);
+  public:
+    /// Create MaterialX data constainer.
+    /// The element path and document that the element resides in are passed in
+    /// as input arguments
+    MaterialXData(const std::string& materialXDocument, const std::string& elementPath);
 	~MaterialXData();
 
+    /// Returns whether the element set to render is a valid output
 	bool isValidOutput();
+
+    /// Create the OGS XML wrapper for shader fragments associated
+    /// with the element set to render
 	void createXMLWrapper();
+
+    /// Register the fragment(s)
 	void registerFragments();
 
     MaterialXData& operator=(const MaterialXData&) = delete;
     MaterialXData& operator=(MaterialXData&&) = delete;
 
-	MaterialX::FilePath libSearchPath;
-	MaterialX::DocumentPtr doc;
-	MaterialX::ElementPtr element;
+    /// Return MaterialX document 
+    MaterialX::DocumentPtr getDocument() const
+    {
+        return _document;
+    }
+
+    /// Return name of shader fragment
+    const MString& getFragmentName() const
+    {
+        return _fragmentName;
+    }
+
+    /// Retuern pointer to the OGS XML wrapper
+    MaterialX::OGSXMLFragmentWrapper* getFragmentWrapper() const
+    {
+        return _xmlFragmentWrapper.get();
+    }
+
+  protected:
+    MaterialX::FilePath _librarySearchPath;
+    MaterialX::DocumentPtr _document;
+    MaterialX::ElementPtr _element;
 
-	MString fragmentName;
-	std::unique_ptr<MaterialX::OGSXMLFragmentWrapper> glslFragmentWrapper;
-	std::vector<std::unique_ptr<MaterialX::GenContext>> contexts;
+    MString _fragmentName;
+    std::unique_ptr<MaterialX::OGSXMLFragmentWrapper> _xmlFragmentWrapper;
```

Comment:
I got rid of the exposure of this completely since it's not required. Also did some sample docs for this class.
One other style suggesting is to use full names if it makes sense (not to wordy). e.g. doc becomes _document.